contentcuration/contentcuration/assets/channelList/vuex/channelList/actions.js

```diff
@@ -0,0 +1,114 @@
+import { ChannelListUrls, ChannelInvitationMapping } from '../../constants';
+import { isDummyId, getDirtyDiff } from '../../utils';
+import client from 'shared/client';
+
+/* CHANNEL LIST ACTIONS */
+export function loadChannelList(context, listType) {
+  return client.get(ChannelListUrls[listType]).then(response => {
+    const channels = response.data;
+    context.commit('ADD_CHANNELS', channels);
+    return channels;
+  });
+}
+
+export function loadAllChannels(context) {
+  return client.get(window.Urls['channelslim-list']()).then(response => {
+    const channels = response.data;
+    context.commit('ADD_CHANNELS', channels);
+    return channels;
+  });
+}
+
+export function addStar(context, channelId) {
+  context.commit('SET_CHANNEL_BOOKMARK', { id: channelId, bookmark: true });
+  return client
+    .post(window.Urls.add_bookmark(), {
+      channel_id: channelId,
+      user_id: context.rootGetters.currentUserId,
+    })
+    .catch(() => {
+      context.commit('SET_CHANNEL_BOOKMARK', { id: channelId, bookmark: false });
+    });
+}
+
+export function removeStar(context, channelId) {
+  context.commit('SET_CHANNEL_BOOKMARK', { id: channelId, bookmark: false });
+  return client
+    .post(window.Urls.remove_bookmark(), {
+      channel_id: channelId,
+      user_id: context.rootGetters.currentUserId,
+    })
+    .catch(() => {
+      context.commit('SET_CHANNEL_BOOKMARK', { id: channelId, bookmark: true });
+    });
+}
+
+/* CHANNEL EDITOR ACTIONS */
+export function saveChannel(context, channelId) {
+  if (context.getters.getChannelIsValid(channelId)) {
+    const channelData = getDirtyDiff(context.getters.getChannel(channelId));
```

Comment:
Get diff here - this means that we no longer bootstrap specific info into the default channel for a new channel, and it gets added here if needed (otherwise it would not show up on the diff).